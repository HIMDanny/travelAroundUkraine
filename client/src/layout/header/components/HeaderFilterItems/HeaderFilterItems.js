/* eslint-disable arrow-body-style */
import React from 'react';
import { Box, Paper, MenuList, MenuItem, Typography } from '@mui/material';
import { styled } from '@mui/system';

const items = [
    {
        image: '../../assets/images/headerFilter/1.jpeg',
        name: 'karpaty - 1 day + spa',
        price: ' 4000 uah',
        amountOfDay: '1 day',
    },
    {
        image: '../../assets/images/headerFilter/2.jpeg',
        name: 'winter karpaty',
        price: ' 3500 uah',
        amountOfDay: '2 day',
    },
    {
        image: '../../assets/images/headerFilter/3.jpeg',
        name: 'weel in karpaty',
        price: ' 7000 uah',
        amountOfDay: '7 day',
    },
];

const ItemBox = styled((props) => <Box  {...props} />)(() => ({
    paddingLeft: 0,
    display: 'grid',
    gridTemplateColumns: 'auto auto',
    gridTemplateRows: 'repeat(2, 1fr)',
    gridColumnGap: ' 10px',
    gridRowGap: '0px',
    boxSizing: 'border-box'
}));

const HeaderFilterItems= () => {
    return (
        <Paper sx={{ width: '242px', borderRadius: '5px', position: 'absolute'}}>
            <Typography sx={{ paddingTop: '14px', paddingLeft: '10px' }}>found: 3</Typography>
            {items.map(({ image, name, price, amountOfDay }) => (
                <MenuList key={name}>
                    <MenuItem sx={{ paddingLeft: 1 }}>
                        <ItemBox>
                            <Box sx={{ gridArea: '1 / 1 / 3 / 3' }}>
                                <img src={image} alt="karpaty" />
                            </Box>
                            <Box sx={{ gridArea: '1 / 3 / 2 / 5' }}>{name}</Box>
                            <Box sx={{ gridArea: '2 / 3 / 3 / 4' }}>{price}</Box>
                            <Box sx={{ gridArea: '2 / 4 / 3 / 5' }}>{amountOfDay}</Box>
                        </ItemBox>
                    </MenuItem>
                </MenuList>
            ))}
            <Typography align='center'>no results.</Typography>
            <Typography align='center' sx={{ marginBottom: '15px' }}>please, change your request</Typography>
        </Paper>
    )
}

export default HeaderFilterItems;