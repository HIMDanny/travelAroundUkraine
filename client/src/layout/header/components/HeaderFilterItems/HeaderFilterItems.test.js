/* eslint-disable react/prop-types */
import { render } from '@testing-library/react';
import HeaderFilterItems from './HeaderFilterItems';

jest.mock('@mui/material/Box', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/Typography', () => ({ children }) => <p>{children}</p>);
jest.mock('@mui/material/MenuList', () => ({ children }) => <ul>{children}</ul>);
jest.mock('@mui/material/MenuItem', () => ({ children }) => <li>{children}</li>);
jest.mock('@mui/material/Paper', () => ({ children }) => <div>{children}</div>);

describe('HeaderFilterItems snapshot testing', () => {
  test('should render HeaderFilterItems component', () => {
    const { asFragment } = render(<HeaderFilterItems />);
    expect(asFragment()).toMatchSnapshot();
  });
});