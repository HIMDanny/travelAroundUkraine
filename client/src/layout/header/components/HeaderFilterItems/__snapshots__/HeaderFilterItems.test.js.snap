// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderFilterItems snapshot testing should render HeaderFilterItems component 1`] = `
<DocumentFragment>
  <div>
    <p>
      found: 3
    </p>
    <ul>
      <li>
        <div>
          <div>
            <img
              alt="karpaty"
              src="../../assets/images/headerFilter/1.jpeg"
            />
          </div>
          <div>
            karpaty - 1 day + spa
          </div>
          <div>
             4000 uah
          </div>
          <div>
            1 day
          </div>
        </div>
      </li>
    </ul>
    <ul>
      <li>
        <div>
          <div>
            <img
              alt="karpaty"
              src="../../assets/images/headerFilter/2.jpeg"
            />
          </div>
          <div>
            winter karpaty
          </div>
          <div>
             3500 uah
          </div>
          <div>
            2 day
          </div>
        </div>
      </li>
    </ul>
    <ul>
      <li>
        <div>
          <div>
            <img
              alt="karpaty"
              src="../../assets/images/headerFilter/3.jpeg"
            />
          </div>
          <div>
            weel in karpaty
          </div>
          <div>
             7000 uah
          </div>
          <div>
            7 day
          </div>
        </div>
      </li>
    </ul>
    <p>
      no results.
    </p>
    <p>
      please, change your request
    </p>
  </div>
</DocumentFragment>
`;
